### DISCUSSION TITLE
Dark mode in a monorepo doesn't work as expected

### DISCUSSION THREAD

**Discussion by idelice** [tailwindlabs/tailwindcss#11262] [Help] [answered]
I have the following folder structure for my monorepo using yarn workspaces:

```
apps
-web (nextjs app)

packages
-ui-lib (react, ts, rollup)
-tailwind-config

package.json
```

both ui lib and my app has tailwind css installed and works fine adding class names and some styling. 

The problem occurs when I try to implement dark mode. 

My config tailwind-config contains following:

```
module.exports = {
  darkMode:"class",
  content: [
    // app content
    `src/**/*.{js,ts,jsx,tsx}`,
    // include packages if not transpiling
    // "../../packages/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {},
  plugins: [],
};
```

And I consume this generic config in both tailwind configs inside ui-lib and nextjs like so:
```
const sharedConfig = require("@tailwind-config/tailwind.config.js");

module.exports = {
  presets: [sharedConfig],
};
```

I have a toggler for switching between dark and light mode - and also updating the localstorage like the example you have in your docs:

```
import { useEffect, useState } from "react";

export const useDarkMode = () => {
  const [isDark, setDarkMode] = useState(false);

  useEffect(() => {
    updateTheme();
  }, []);


  const handleDarkMode = (mode: "dark" | "light") => {
    localStorage.setItem("theme", mode);
    updateTheme();
  };

  const updateTheme = () => {
    if (
      localStorage.theme === "dark" ||
      (!("theme" in localStorage) &&
        window.matchMedia("(prefers-color-scheme: dark)").matches)
    ) {
      document.documentElement.classList.add("dark");

      setDarkMode(true)
    } else {
      document.documentElement.classList.remove("dark");

      setDarkMode(false)
    }
  };

  return {
    handleDarkMode,
    isDark,
  };
};
```

And as I said, it works perfectly fine for all my styling in my nextjs app. 

**_So the question is, why does dark mode _not_ work inside my ui-lib?_**

**Comment by idelice** [comment, marked as answer]
Seems like I fixed the issue after hours of struggling with a bit of help from this generous man from Stackoverflow: https://stackoverflow.com/questions/71461174/tailwind-dark-mode-does-not-working-with-the-prefix-why 

Please document this. I could not find anywhere in your docs that says you'll have to update the `darkMode` property to an array of multiple values to fix issues with prefixed tailwind css. 

This is especially useful information when working on monorepo's.

**Reply by skylarkuu** [reply]
@Delice0  I am using the turbo repo and I have the same problem. Can you please share your tailwind configs with your fixes?
